src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -679,7 +704,7 @@ public override int GetMethodMetadataToken(MethodInfo method)
                 return MetadataTokens.GetToken(mb._handle);
             }
 
-            return 0;
+            return method.MetadataToken;
```

Comment:
Is this a break change or a bug that needed to be fixed?

Reply:
> Is this a break change or a bug that needed to be fixed?

I thought returning `method.MetadataToken` might be useful in case the method doesn't belong to the module, but not sure if that scenario should be covered, anyway now I see the condition above need to be changed to exclude default case.